core/src/main/scala/fs2/Async.scala

```diff
@@ -189,6 +189,13 @@ object Async {
    */
   case class Change[+A](previous: A, now: A)
 
+  object Change {
+    implicit class ChangeSyntax[A](val self:Change[A]) extends AnyVal {
```

Comment:
Performance wise, they are likely equivalent -- monomorphic method on the class vs a monomorphic method on the companion. Neither will result in object allocation. I'd avoid the syntax style unless you need it to work around variance -- e.g., defining methods that fix a covariant/contravariant type param to invariant.